packages/components/src/Tokens/TokensAccountVerify.tsx

```diff
@@ -15,53 +17,70 @@ import { colors } from "../styleConstants";
-        </UserTokenAccountRequirement>
-      </FlexColumnsPrimaryModule>
-    );
-  } else if (step === TOKEN_PROGRESS.ACTIVE) {
-    return (
-      <FlexColumnsPrimaryModule padding={true}>
-        <TokenAccountSectionHeader>
-          <TokenVerifySectionText />
-        </TokenAccountSectionHeader>
-        <UserTokenAccountRequirement step={step}>
-          <TokenRequirementIcon step={step}>
-            <CivilTutorialIcon />
-          </TokenRequirementIcon>
-          <TokenQuizSectionText />
-          <TokenBtns onClick={props.handleOpen}>
-            <TokenQuizBtnText />
-          </TokenBtns>
-          <FullScreenModal open={props.open || false} solidBackground={true}>
-            <CloseBtn onClick={props.handleClose}>
-              <CloseXIcon color={colors.accent.CIVIL_GRAY_2} />
-            </CloseBtn>
-            <TokenTutorial handleClose={props.handleClose} />
-          </FullScreenModal>
-        </UserTokenAccountRequirement>
-      </FlexColumnsPrimaryModule>
-    );
+    if (step === TOKEN_PROGRESS.DISABLED) {
+      return (
+        <FlexColumnsPrimaryModule padding={true}>
+          <TokenAccountSectionHeader>
+            <TokenVerifySectionText />
+          </TokenAccountSectionHeader>
+          <UserTokenAccountRequirement step={step}>
+            <TokenRequirementIcon step={step}>
+              <CivilTutorialIcon color={colors.accent.CIVIL_GRAY_3} />
+            </TokenRequirementIcon>
+            <TokenQuizSectionText />
+          </UserTokenAccountRequirement>
+        </FlexColumnsPrimaryModule>
+      );
+    } else if (step === TOKEN_PROGRESS.ACTIVE) {
+      return (
+        <FlexColumnsPrimaryModule padding={true}>
+          <TokenAccountSectionHeader>
+            <TokenVerifySectionText />
+          </TokenAccountSectionHeader>
+          <UserTokenAccountRequirement step={step}>
+            <TokenRequirementIcon step={step}>
+              <CivilTutorialIcon />
+            </TokenRequirementIcon>
+            <TokenQuizSectionText />
+            <Query query={getCurrentUserQuery}>
+              {({ loading, error, data }) => {
+                const quizPayload = loading || error ? {} : data.currentUser.quizPayload;
+
+                return (
+                  <TokenBtns onClick={handleOpen}>
+                    {this.isQuizStarted(quizPayload) ? "Continue" : <TokenQuizBtnText />}
+                  </TokenBtns>
+                );
+              }}
+            </Query>
+            <FullScreenModal open={open || false} solidBackground={true}>
+              <CloseBtn onClick={handleClose}>
+                <CloseXIcon color={colors.accent.CIVIL_GRAY_2} />
+              </CloseBtn>
+              <TokenTutorial handleClose={handleClose} />
+            </FullScreenModal>
+          </UserTokenAccountRequirement>
+        </FlexColumnsPrimaryModule>
+      );
+    }
+
+    return <></>;
   }
 
-  return <></>;
-};
+  private isQuizStarted = (quizPayload: {}) => {
```

Comment:
This could just be `return Object.keys(quizPayload).length > 0`.

(or even `return Object.keys(quizPayload).length`, but I like the explicit `> 0`).

Reply:
👍 good call